### DISCUSSION TITLE
Use FFI for consuming external libraries (Rust and C++)

### DISCUSSION THREAD

**Discussion by InDIOS** [alshdavid/TypeScriptBC#65] [Ideas]
Would be a great idea to use FFI to communicate with Rust and C++ libraries, like Deno does. This could be easy to use and easy to understand for beginners, depending of the implementation you choose.

**Comment by marcus-sa** [comment]
WASM?

**Comment by SuperSonicHub1** [comment]
WASM FFI could actually be really cool; see https://wasmer.io/.

**Comment by alshdavid** [comment]
For sure, sounds like a great way to dynamically access external functionality - [can follow a similar approach to Deno](https://deno.land/manual/runtime/ffi_api). 

A WASM  FFI would be very interesting for sure!
